lib/libzfs/libzfs_import.c

```diff
@@ -1675,6 +1675,45 @@ zpool_clear_label(int fd)
 	return (0);
 }
 
+/*
+ * Scan a list of devices for zfs devices.
+ */
+static int
+zpool_find_import_path(libzfs_handle_t *hdl, kmutex_t *lock,
+    avl_tree_t **slice_cache, char **path, int paths)
+{
+	avl_tree_t *cache;
+	rdsk_node_t *slice;
+	int i, error;
+
+	*slice_cache = NULL;
+	cache = zfs_alloc(hdl, sizeof (avl_tree_t));
```

Comment:
As written this function can't fail so you can unconditionally assign `slice_cache` here and drop the local `cache` variable.